lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -84,6 +94,49 @@ class SPIRVRegularizeLLVM : public ModulePass {
 
 char SPIRVRegularizeLLVM::ID = 0;
 
+void SPIRVRegularizeLLVM::lowerMemset(MemSetInst *MSI) {
+  if (isa<Constant>(MSI->getValue()) && isa<ConstantInt>(MSI->getLength()))
+    return; // To be handled in LLVMToSPIRV::transIntrinsicInst
+  Function *IntrinsicFunc = MSI->getCalledFunction();
+  assert(IntrinsicFunc && "Missing function");
+  std::string FuncName = IntrinsicFunc->getName().str();
+  std::replace(FuncName.begin(), FuncName.end(), '.', '_');
+  FuncName = "spirv." + FuncName;
+  if (MSI->isVolatile())
+    FuncName += ".volatile";
+
+  // Redirect @llvm.memset.* call to @spirv.llvm_memset_*
+  Function *F = M->getFunction(FuncName);
+  if (F) {
+    // This function is already linked in.
+    MSI->setCalledFunction(F);
+    return;
+  }
+  // TODO copy arguments attributes: nocapture writeonly.
```

Comment:
Are you still planning to do this as part of this PR?

Reply:
No, I'd prefer to do it in a separate PR.